MathLib/Nonlinear/Root1D.h

```diff
@@ -0,0 +1,174 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef MATHLIB_NONLINEAR_ROOT1D_H
+#define MATHLIB_NONLINEAR_ROOT1D_H
+
+#include <cassert>
+#include <cmath>
+
+namespace MathLib
+{
+namespace Nonlinear
+{
+
+namespace detail
+{
+
+inline
+bool same_sign(const double a, const double b)
+{
+    return (a < 0.0) == (b < 0.0);
```

Comment:
So if a = -0 and b = 1, same_sign(-0, 1) = false, that's exactly the signbit(a) == signbit(b)...